core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/dao/OrderDaoImpl.java

```diff
@@ -72,7 +72,9 @@ public class OrderDaoImpl implements OrderDao {
 
     @Override
     public Order readOrderById(final Long orderId) {
-        return em.find(OrderImpl.class, orderId);
+        TypedQuery<Order> query = em.createQuery("SELECT o FROM OrderImpl o WHERE o.id=?1", Order.class);
```

Comment:
My note is only about syntax. Maybe better to use something like this?

>TypedQuery<Order> query = em.createQuery("SELECT o FROM OrderImpl o WHERE o.id=:orderId", Order.class);
        query.setParameter("orderId", orderId);